inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.cpp

```diff
@@ -3,155 +3,137 @@
-//     */
-//    SizeVector weightDims;
-//    if (withGroups) {
-//        weightDims = {
-//                deconvLayer->_group,
-//                deconvLayer->input()->getTensorDesc().getDims()[1] / deconvLayer->_group,
-//                deconvLayer->_out_depth / deconvLayer->_group,
-//        };
-//        groupNum = deconvLayer->_group;
-//    } else {
-//        weightDims = {
-//                deconvLayer->input()->getTensorDesc().getDims()[1],
-//                deconvLayer->_out_depth
-//        };
-//    }
-//    for (int i = 1; i <= deconvLayer->_kernel.size(); i++) {
-//        weightDims.push_back(deconvLayer->_kernel[deconvLayer->_kernel.size() - i]);
-//    }
-//
-//    if (getParentEdges().size() == 1)
-//        internalBlobs.push_back(createInternalBlob(weightDims, true));
-//
-//    invertVectorCopyUtoI(deconvLayer->_stride, stride);
-//    for (int i = 1; i <= deconvLayer->_dilation.size(); i++) {
-//        dilation.push_back(static_cast<int>(deconvLayer->_dilation[deconvLayer->_dilation.size() - i]) - 1);
-//    }
-//    auto allPads = getPaddings(*deconvLayer);
-//    invertVectorCopyUtoI(allPads.begin, paddingL);
-//    invertVectorCopyUtoI(allPads.end, paddingR);
-//
-//    weightsDims = MKLDNNDims(weightDims);
-//
-//    for (int i = 0; i < paddingR.size(); i++) {
-//        int with_group = (withGroups) ? 1 : 0;
-//        int krn = weightsDims[with_group + 2 + i];
-//        int src = getChildEdgeAt(0)->getDims()[2 + i];
-//        int dst = getParentEdgeAt(0)->getDims()[2 + i];
-//
-//        krn = (krn - 1)*(dilation[i] + 1) + 1;
-//        int calc_dst = (src - krn + paddingL[i]) / stride[i] + 1;
-//        paddingR[i] = (dst - calc_dst) * stride[i];
-//    }
-//
-//    for (auto format : getAvailableFormatsForDims(getParentEdgeAt(0)->getDims())) {
-//        MKLDNNMemoryDesc in_candidate(getParentEdgeAt(0)->getDims(), inputDataType, format);
-//        MKLDNNMemoryDesc out_candidate(getChildEdgeAt(0)->getDims(), outputDataType, format);
-//        createDescriptor({in_candidate}, {out_candidate});
-//    }
+    if (!descs_fwd.empty() && !descs_bwd.empty())
+        return;
+
+    InferenceEngine::Precision precision = getOriginalInputPrecisionAtPort(0);
+    if (!one_of(precision, InferenceEngine::Precision::FP32, InferenceEngine::Precision::BF16))
+        precision = InferenceEngine::Precision::FP32;
+    auto inputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+    precision = getOriginalOutputPrecisionAtPort(0);
+    if (!one_of(precision, InferenceEngine::Precision::FP32, InferenceEngine::Precision::BF16))
+        precision = InferenceEngine::Precision::FP32;
+    auto outputDataType = MKLDNNExtensionUtils::IEPrecisionToDataType(precision);
+    if (inputDataType == memory::data_type::bf16 || outputDataType == memory::data_type::bf16)
+       inputDataType = outputDataType = memory::data_type::bf16;
+
+    if (getParentEdges().size() != 2 && getParentEdges().size() != 3)
```

Comment:
Seems like `getParentEdges().size() == 3` is not valid case anymore since we don't fuse bias into Deconvolution on the plug-in level (unlike for Convolution).

Reply:
in specs says that ConvolutionBackpropData and GroupConvolutionBackpropData can have optional third input